src/main/java/pau/task/Task.java

```diff
@@ -0,0 +1,75 @@
+package pau.task;
+
+/**
+ * Represents a task.
+ */
+public abstract class Task {
+
+    /**
+     * Description of the task.
+     */
+    protected String description;
+
+    /**
+     * Status of task completion.
+     */
+    protected boolean isDone;
+
+    /**
+     * Constructs a task.
+     *
+     * @param description Description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Writes the contents of the task to the hard disk.
+     *
+     * @return The contents of the task that is saved to the hard disk.
+     */
+    public abstract String writeToFile();
```

Comment:
I think the description of this method may be misleading? From what I have read in the `ToDo`, `Deadline` and `Event` implementation of the method, it is simply returning the storage format in a `String` and not actually writing to the file? Can consider renaming the method to `getStorageFormat` and update the description to make it clearer?